cvmfs/swissknife_graft.cc

```diff
@@ -0,0 +1,249 @@
+/**
+ * This file is part of the CernVM File System.
+ *
+ * Process a set of input files and create appropriate graft files.
+ */
+
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/stat.h>
+
+#include <cstdio>
+#include <iostream>
+#include <vector>
+#include <sstream>
+
+#include "compression.h"
+#include "fs_traversal.h"
+#include "hash.h"
+
+#include "swissknife_graft.h"
+
+#define CHUNK_SIZE 16*1024
+
+int swissknife::CommandGraft::ChecksumFd(int fd, shash::Any &id, off_t &processed_size) {  // NOLINT
```

Comment:
Ah, I see that the processed size is needed as well.  I think it would be nicer to extend `CompressFd2Null()` with an `uint64_t` output pointer that defaults to `NULL`.